Providers/SDWebThumbnails.swift

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Storage
+
+/**
+ * SDWebImage implementation of the Thumbnails protocol.
+ */
+public class SDWebThumbnails: Thumbnails {
+    let cache: SDImageCache
+
+    required public init(files: FileAccessor) {
+        self.cache = SDImageCache(namespace: "thumbnails")
+    }
+
+    public func clear(complete: ((success: Bool) -> Void)?) {
+        cache.clearMemory()
+        cache.clearDiskOnCompletion { () -> Void in
+            dispatch_async(dispatch_get_main_queue()) {
+                complete?(success: true)
+                return
+            }
+        }
+    }
+
+    public class func getKey(url: NSURL) -> String {
+        return getKey(url.absoluteString!)
+    }
+
+    public class func getKey(url: String) -> String {
+        return "thumbnail://\(url)"
```

Comment:
I was going to say it might be a good idea to continue using SHA1 hashing here in case there's a long URL (or even a data URI), but since `SDImageCache` uses `NSCache` under the covers, maybe this is automatically done for us.